# Standard Error, Margin of Error and Confidence Interval

Statement:
Here is a breakdown of SPX rets on the week categorized by the week relative to Opex. Recently 2 weeks prior to Opex (e.g. 7/30/21 to 8/6/21 in this last cycle) has been extremely bullish.


**Question:**

Assume a simple world where returns are i.i.d. normally distributed with 18% annualized volatility. 

What is the standard error of the trailing 3-year average weekly return for each week relative to expiration?


In [13]:
import pandas as pd
data = pd.read_csv("opex_ret.csv")
data

opex_date  year month  opex_wk  1wk_to_opex  2wk_to_opex  3wk_to_opex
0    1/17/20  2020   Jan   0.0197       0.0094      -0.0016       0.0058
1    2/21/20  2020   Feb  -0.0125       0.0158       0.0317      -0.0212
2    3/20/20  2020   mar  -0.1498      -0.0879       0.0061      -0.1149
3    4/17/20  2020   Apr   0.0304       0.1210      -0.0208       0.1026
4    5/15/20  2020   May  -0.0226       0.0350      -0.0021      -0.0132
5    6/19/20  2020   Jun   0.0186      -0.0478       0.0491       0.0301
6    7/17/20  2020   Jul   0.0125       0.0176       0.0402      -0.0286
7    8/21/20  2020   Aug   0.0072       0.0064       0.0245       0.0173
8    9/18/20  2020   Sep  -0.0064      -0.0251      -0.0231       0.0326
9   10/16/20  2020   Oct   0.0019       0.0384       0.0152      -0.0063
10  11/20/20  2020   Nov  -0.0077       0.0216       0.0732      -0.0564
11  12/18/20  2020   Dec   0.0125      -0.0096       0.0167       0.0227
12   1/15/21  2021   Jan   0.0072       0.0183       0.0143      -0.0017
13   2/19/21  2021   Feb  -0.0064       0.0123       0.0465      -0.0331
14   3/19/21  2021   Mar   0.0019       0.0264       0.0081      -0.0245
15   4/16/21  2021   Apr  -0.0077       0.0271       0.0114       0.0157
16   5/21/21  2021   May   0.0137      -0.0139       0.0123       0.0002
17   6/18/21  2021   Jun  -0.0043       0.0041       0.0061       0.0116
18   7/16/21  2021   Jul  -0.0191       0.0040       0.0167       0.0274

**Is 2 week prior to op-ex Extremely bullish?**

If the returns are extremely bullish, the mean weekly return of the 2-week prior to opex must be larger than the mean weekly return of other distributions - op-ex week, 1-week prior to opex, and 3-wk prior to opex. That is indeed the case, as seen below. 

SPX returns 1.7% 2 weeks prior to opex, which is obviously higher in comparison to -0.58% during opex week, 0.91% in a week prior to opex, and -0.18% 3 weeks prior to opex. 

In [3]:
from IPython.core.display import display, HTML
desc = data.describe()
desc.drop(['count'], inplace=True)
desc.drop(columns='year', inplace=True)
output = desc.to_html(formatters={
    'opex_wk': '{:,.2%}'.format,
    '1wk_to_opex': '{:,.2%}'.format,
    '2wk_to_opex': '{:,.2%}'.format,
    '3wk_to_opex': '{:,.2%}'.format
})
display(HTML(output))

**Standard error of the trailing 3-year average weekly return**

The returns are weekly but the given standard deviation is annualized. If annualized vol or annualized standard deviation is 18%, weekly vol = 2.5% (18% / sqrt(52 weeks))

The standard error of the trailing 3-year average weekly return is therefore, 

$$ \frac{\sigma} {\sqrt n} $$

a. $\sigma$ is population standard deviation. Because population standard deviation is unknown in most of the cases, unlike for our simple world, an unbiased estimate of the standard error of a mean can be obtained by dividing the sample standard deviation (s) by the square root of the number of observations in that sample.

b. n is the number of observations

In our world, there are 12 opex in a year, so therefore, the total number of weekly expirations in 3 years = 36.  

<br><br>
Therefore the standard error of the trailing 3-year average weekly return = $\frac{2.5} {\sqrt 36}$ = 0.42%


**Confidence Interval and Margin of Error**

The 95% confidence interval for the 2wk_to_opex is [1.71% ± 1.96×0.42%] = [0.89, 2.53]

Although the margin of error is quite wide (0.82%), the weekly returns of 2 week prior to opex is still substantially greater than the true mean of ~0.23% per week (12% annual returns/52 weeks)  


<hr>
**References**

1. Inspired by https://twitter.com/pat_hennessy/status/1428875608046542848